tests/e2e/tests/simple/simple1_test.go

```diff
@@ -148,6 +152,82 @@ func TestAuth(t *testing.T) {
 	}
 }
 
+func Test404sDuringChanges(t *testing.T) {
+	url := "http://" + tc.Kube.Ingress + "/fortio/debug"
+	rulePath1 := util.GetResourcePath(routingR1Yaml)
+	rulePath2 := util.GetResourcePath(routingR2Yaml)
+	go func() {
+		time.Sleep(9 * time.Second)
+		glog.Infof("Changing rules mid run to v1/v2")
+		if err := tc.Kube.Istioctl.CreateRule(rulePath1); err != nil {
+			t.Errorf("istioctl rule create %s failed", routingR1Yaml)
+			return
+		}
+		time.Sleep(4 * time.Second)
+		glog.Infof("Changing rules mid run to a/b")
+		if err := tc.Kube.Istioctl.CreateRule(rulePath2); err != nil {
+			t.Errorf("istioctl rule create %s failed", routingR1Yaml)
+			return
+		}
+		time.Sleep(4 * time.Second)
+		util.KubeDelete(tc.Kube.Namespace, rulePath1) // nolint:errcheck
+		util.KubeDelete(tc.Kube.Namespace, rulePath2) // nolint:errcheck
+	}()
+	// run at a low/moderate QPS for a while while changing the routing rules,
+	// check for any non 200s
+	opts := fhttp.HTTPRunnerOptions{
+		RunnerOptions: periodic.RunnerOptions{
+			QPS:        8,
+			Duration:   20 * time.Second,
+			NumThreads: 8,
+		},
+		URL: url,
+	}
+	res, err := fhttp.RunHTTPTest(&opts)
+	if err != nil {
+		t.Fatalf("Generating traffic via fortio failed: %v", err)
+	}
+	numRequests := res.DurationHistogram.Count
+	num200s := res.RetCodes[http.StatusOK]
+	if num200s != numRequests {
+		t.Errorf("Not all %d requests were successful (%v)", numRequests, res.RetCodes)
+	}
+}
+
+func Test503sWithPartialRules(t *testing.T) {
+	url := "http://" + tc.Kube.Ingress + "/fortio/debug"
+	rulePath := util.GetResourcePath(routingRNPYaml)
+	go func() {
+		time.Sleep(9 * time.Second)
+		glog.Infof("Changing rules with some non existent destination, mid run")
+		if err := tc.Kube.Istioctl.CreateRule(rulePath); err != nil {
+			t.Errorf("istiocrl create rule %s failed", routingRNPYaml)
+			return
+		}
+	}()
+	defer tc.Kube.Istioctl.DeleteRule(rulePath) // nolint:errcheck
+	// run at a low/moderate QPS for a while while changing the routing rules,
+	// check for any non 200s
+	opts := fhttp.HTTPRunnerOptions{
+		RunnerOptions: periodic.RunnerOptions{
+			QPS:        8,
+			Duration:   20 * time.Second,
+			NumThreads: 8,
+		},
+		URL: url,
+	}
+	res, err := fhttp.RunHTTPTest(&opts)
+	if err != nil {
+		t.Fatalf("Generating traffic via fortio failed: %v", err)
+	}
+	numRequests := res.DurationHistogram.Count
+	num200s := res.RetCodes[http.StatusOK]
+	// 1 or a handful of 503s is maybe ok, but not 1/2
+	if num200s != numRequests {
```

Comment:
Based on comment, should these be compared directly or should it do something like `if num200s != (numRequests * 0.95) { ...}` or similar (i.e. check that some % of requests were successful, but not 100%).

Reply:
it's a bit of the feature/requirement question : should changing rules for existing already deployed/up svc create any error ? in the strictest sense I think it shouldn't - there should be 0 errors, but maybe a handful is ok, my comment is somewhat outdated or reflecting what I observed (a lot of 503s)